src/test/app/Credentials_test.cpp

```diff
@@ -0,0 +1,949 @@
+            }
+        }
+
+        {
+            testcase("Delete  by issuer");
+            env(credentials::create(subject, issuer, credType));
+            env.close();
+
+            env(credentials::deleteCred(issuer, subject, issuer, credType));
+            env.close();
+            {
+                auto const credKey =
+                    credentialKeylet(subject, issuer, credType);
+                BEAST_EXPECT(!env.le(credKey));
+                BEAST_EXPECT(!ownerCount(env, subject));
+                BEAST_EXPECT(!ownerCount(env, issuer));
+                auto const jle =
+                    credentials::ledgerEntry(env, subject, issuer, credType);
+                BEAST_EXPECT(
+                    jle.isObject() && jle.isMember(jss::result) &&
+                    jle[jss::result].isMember(jss::error) &&
+                    jle[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+    }
+
+    void
+    testCreateFailed(FeatureBitset features)
+    {
+        using namespace test::jtx;
+
+        const char credType[] = "abcde";
+
+        Account const issuer{"issuer"};
+        Account const subject{"subject"};
+
+        {
+            using namespace jtx;
+            Env env{*this, features};
+
+            env.fund(XRP(5000), subject, issuer);
+            env.close();
+
+            {
+                testcase("Credentials fail, no subject param.");
+                auto jv = credentials::create(subject, issuer, credType);
+                jv.removeMember(jss::Subject);
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                auto jv = credentials::create(subject, issuer, credType);
+                jv[jss::Subject] = to_string(xrpAccount());
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                testcase("Credentials fail, no credentialType param.");
+                auto jv = credentials::create(subject, issuer, credType);
+                jv.removeMember(sfCredentialType.jsonName);
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                testcase("Credentials fail, empty credentialType param.");
+                auto jv = credentials::create(subject, issuer, "");
+                env(jv, ter(temMALFORMED));
+            }
+
+            {
+                testcase(
+                    "Credentials fail, credentialType length > "
+                    "maxCredentialTypeLength.");
+                constexpr std::string_view longCredType =
+                    "abcdefghijklmnopqrstuvwxyz01234567890qwertyuiop[]"
+                    "asdfghjkl;'zxcvbnm8237tr28weufwldebvfv8734t07p   "
+                    "9hfup;wDJFBVSD8f72  "
+                    "pfhiusdovnbs;"
+                    "djvbldafghwpEFHdjfaidfgio84763tfysgdvhjasbd "
+                    "vujhgWQIE7F6WEUYFGWUKEYFVQW87FGWOEFWEFUYWVEF8723GFWEFB"
+                    "WULE"
+                    "fv28o37gfwEFB3872TFO8GSDSDVD";
```

Comment:
Can this be a shorter testcase so that it's actually testing the 64 byte limit?

Reply:
fixed